src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -348,18 +348,92 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
                 if disk_version.lower() == 'latest':
                     disk_version = _get_latest_image_version(cmd.cli_ctx, location, disk_publisher, disk_offer,
                                                              disk_sku)
-                client = _compute_client_factory(cmd.cli_ctx)
-                response = client.virtual_machine_images.get(location, disk_publisher, disk_offer, disk_sku,
-                                                             disk_version)
-                image_reference = response.id
             else:  # error
                 raise CLIError('usage error: --image-reference should be ID or URN (publisher:offer:sku:version).')
+        else:
+            from msrestazure.tools import parse_resource_id
+            terms = parse_resource_id(image_reference)
+            disk_publisher, disk_offer, disk_sku, disk_version = \
+                terms['child_name_1'], terms['child_name_3'], terms['child_name_4'], terms['child_name_5']
+
+        client = _compute_client_factory(cmd.cli_ctx)
+        response = client.virtual_machine_images.get(location, disk_publisher, disk_offer, disk_sku,
+                                                     disk_version)
+
+        if hyper_v_generation and hyper_v_generation == 'V1' and not image_reference_lun:
+            logger.warning('Please consider upgrading security for your VM resources by using Gen 2 OS Disk and '
+                           'Trusted Launch security type. Please use "--hyper-v-generation V2" to set the '
+                           'hypervisor generation as Gen2 for OS disk. To know more about Trusted Launch, '
+                           'please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
```

Comment:
The judgment condition needs to be added with `response.hyper_v_generation == 'V2'`, otherwise it may be printed repeatedly with the following logs